crates/core/tedge/src/cli/diag/collect.rs

```diff
@@ -52,31 +55,48 @@ impl Command for DiagCollectCommand {
 
         let mut skipped_count = 0;
         let mut error_count = 0;
+        let mut log_buffer = Vec::new(); // To avoid output while spinner is running
 
         for plugin in plugins {
             let banner = format!("Executing {plugin}");
-            let spinner = Spinner::start(banner);
+            let spinner = Spinner::start(banner.clone());
             let res = self.execute_diag_plugin(&plugin).await;
 
             match spinner.finish(res) {
-                Ok(exit_status) if exit_status.success() => {}
-                Ok(exit_status) if exit_status.code() == Some(2) => {
-                    skipped_count += 1;
-                    info!("{plugin} is marked skipped");
-                }
                 Ok(exit_status) => {
-                    error_count += 1;
-                    error!("{plugin} failed with exit status: {exit_status}");
+                    log_buffer.push(format!("{banner}... ✓"));
```

Comment:
While testing, I noticed something weird. The ✔️  signs were not captured/rendered properly. Instead, the output was as follows:

```
Executing /usr/share/tedge/diag-plugins/01_tedge.sh... ?~\~S
Executing /usr/share/tedge/diag-plugins/02_os.sh... ?~\~S
Executing /usr/share/tedge/diag-plugins/03_mqtt.sh... ?~\~S
...
```

Reply:
The file contains the correct utf-8 characters, I think it is just the vim which isn't showing property.

You can tell vim to use utf-8 character by entering the command `set encoding=utf-8`

Reply:
@reubenmiller Yeah, it was vim not detecting the encoding properly. Working fine, once it was set explicitly.